Update metallicity yml to add Fe(SNIa)/H vs Mstellar plot

correac opened the pull request.

bwvdnbro commented:
@correac can you run the formatting script?

EvgeniiChaikin commented:
Hi @correac, could you please explain in the description in more detail why we want this update? 

Also, I have seen that you have recently updated the preprocessing script. Is that update related to the update here?

correac commented:
Formatting done. Yes, the updates here and preprocessing are to include mass-weighted [Fe(from SNIa)/H] vs Mstellar plot.

EvgeniiChaikin commented on colibre/auto_plotter/metallicity.yml:
```diff
@@ -666,20 +665,21 @@ stellar_mass_star_fe_over_h_mol_hifloor_50:
   type: "scatter"
   legend_loc: "upper left"
-  comment: "LoM"
+  comment: "MoL w. [F/H]=-4 floor"
```
You probably want here
```suggestion
  comment: "MoL w. [F/H]=-3 floor"
```

EvgeniiChaikin reviewed (changes requested):
Thank you for the update @correac!

All looks good. I have tested this branch on one of my runs and it works as expected. 

I have left one comment in the code to fix a typo; otherwise, this brunch is ready to be merged. 

@bwvdnbro this branch needs the latest version of the `velociraptor-python` library. Do we want to add a new version (tag) to `velociraptor-python` so that we can specify it here in the requirements? The alternative way would be to release a new version of `velociraptor-python `after the SOAP update is merged.

correac commented:
Fixed! I have also added a few additional changes. I would like to have the datasets of Zahid+2017 and Kudritzki+2016 in the stellar mass - metallicity plots. I have made a pull request in the velociraptor-comparison-data repository to include them.

EvgeniiChaikin commented:
Thank you @correac! Let's (i) first merge that PR with the observational data, (ii) then update the version of the observational-data submodule in this PR, and (iii) then merge this PR.

EvgeniiChaikin commented on colibre/auto_plotter/metallicity.yml:
```diff
@@ -515,8 +514,9 @@ stellar_mass_star_metallicity_50:
     - filename: GalaxyStellarMassStellarMetallicity/Gallazzi2005_Data.hdf5
     - filename: GalaxyStellarMassStellarMetallicity/Kirby2013_Data.hdf5
     - filename: GalaxyStellarMassStellarMetallicity/Fraser-McKelvie_2021.hdf5 
-    - filename: GalaxyStellarMassStellarMetallicity/FIREbox.hdf5
```
Is there any reason why you removed the FIREbox data in this line and in a few other lines?

correac replied:
Too many lines/dots makes it difficult to check our results with obs. data. I left the Firebox in other plots.

EvgeniiChaikin replied:
I agree that now there are a bit too many data points I don't think we should remove FIREbox. FIREbox is one of the main competitors for COLIBRE, so it is of great interest.

correac replied:
It's of interest, I agree. But please let's have it in the other plots, not in the Z-Mstellar nor [Fe/H](mass-weighted)-Mstellar. Also, I will add another dataset from MANGA to these plots next week to better assess our trend with observations. So it will get busy :)

bwvdnbro commented:
> @bwvdnbro this branch needs the latest version of the `velociraptor-python` library. Do we want to add a new version (tag) to `velociraptor-python` so that we can specify it here in the requirements? The alternative way would be to release a new version of `velociraptor-python `after the SOAP update is merged.

I prefer releasing a new version after the SOAP update has been merged. Otherwise this release would be very small and we would be making two releases in a short time.

EvgeniiChaikin commented:
@correac I have just merged the associated PR in the velociraptor-data-comparison repository. Could you please update its version in this PR so that when the user does `git submodule update --init --recursive`, they will be able to generate the data that you have just added?

correac commented:
Done. I updated the data folder.
